app/Components/TreeInventory/index.js

```diff
@@ -22,34 +23,58 @@ import {
   auth0Login,
 } from '../../Actions';
 import { store } from '../../Actions/store';
-import { Colors } from '_styles';
+import { Colors, Typography } from '_styles';
 import { LocalInventoryActions } from '../../Actions/Action';
 import { empty_inventory_banner } from '../../assets';
 import { SvgXml } from 'react-native-svg';
 import moment from 'moment';
 import i18next from 'i18next';
+import NetInfo from '@react-native-community/netinfo';
 
 const TreeInventory = ({ navigation }) => {
   const { dispatch } = useContext(store);
 
   const [allInventory, setAllInventory] = useState(null);
   const [isLoaderShow, setIsLoaderShow] = useState(false);
+  const [mobileData, setMobileData] = useState(false);
+  const [connectionType, setConnectionType] = useState(null);
 
   useEffect(() => {
     const unsubscribe = navigation.addListener('focus', () => {
       initialState();
     });
-
+    // wifiConnection();
     return unsubscribe;
   }, [navigation]);
 
+  useEffect(() => {
+    wifiConnection();
+    wifiUpload();
+    return () => {
+      wifiConnection();
+    };
+  }, [connectionType]);
+
   const onPressInventory = (item) => {
     setTimeout(() => {
       dispatch(LocalInventoryActions.setInventoryId(item.inventory_id));
       navigation.navigate(item.last_screen);
     }, 0);
   };
 
+  const toggleSwitchPublish = () => setMobileData(previousState => !previousState);
+
+  const wifiConnection = () => {
+    NetInfo.addEventListener(state => {
+      setConnectionType(state.type);
+    });
+  };
+
+  const wifiUpload = () => {
+    if (connectionType === 'wifi' && pendingInventory.length > 0) {
```

Comment:
Maybe you should also check state.isConnected before trying to upload? (https://github.com/react-native-community/react-native-netinfo)

Reply:
There is even something as `isInternetReachable` I have not yet tested myself.

Reply:
OK. i will do that. Thanks